src/rust/engine/process_execution/src/lib.rs

```diff
@@ -213,6 +213,21 @@ pub struct ExecuteProcessRequest {
   pub is_nailgunnable: bool,
 }
 
+impl ExecuteProcessRequest {
+  fn user_facing_name(&self) -> String {
```

Comment:
As per this remaining TODO: https://github.com/pantsbuild/pants/blob/b476e17d5ac8122b416d167d92934756e1e8f5b0/src/python/pants/engine/isolated_process.py#L30-L31

Could we do that here? If not a `process_executor` command line directly -- perhaps just a properly-quoted command line string so you can just copy and paste it from the terminal (with env vars at the front like `PEX_IGNORE_RCFILES=true ./pex ...`)?

Reply:
`process_executor` is currently a super pants-developer-facing tool (in fact it doesn't even exist as a published binary - you need to be building it from source in the pants repo in order to use it), and this function is a very user-facing concept - I don't think the two should meet here - rather a `process_executor` command to run may be useful `trace`-level logging? (Can _definitely_ be a separate PR though)

But I'd be in favour of handling quoting and env vars properly for easy copy-and-paste :)

Reply:
A probably-appropriate command line probably starts with `env -i` too.

Reply:
@cosmicexplorer having a faithful command-line representation of `ExecuteProcessRequest` definitely seems like a good idea, but I think that should be part of the `Display` implementation rather than this user-facing name (i.e., implemented in another commit).

Cf. https://github.com/pantsbuild/pants/issues/7509 , we're planning on using user-facing-names of ExecuteProcessRequests in two places right now - workunit reporting and the swim lane display. As @blorente suggests in one of his comments below, it doesn't make sense to optimize the swim lane display for copy-pasting because it changes too fast and will truncate any line to what fits on the screen anyway. I think this is also true for workunit reporting - the workunit for an EPR should tell us that an EPR was running and roughly what it was doing, but we don't need that name to be a copy-pasteable, faithful, and potentially-long command line representation of that command (maybe that should be a metadata item on the workunit, though).

Reply:
Thank you! That explanation is fantastic!